app/models/submission/validations_by_template_name.rb

```diff
@@ -65,5 +80,94 @@ def validate_consistent_column_value(column_header)
       )
     end
   end
-  # rubocop:enable Metrics/MethodLength,Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  def validate_samples_per_pool_for_labware
+    return if headers.index(HEADER_BARCODE).nil? && headers.index(HEADER_PLATE_WELLS).nil?
+
+    grouped_rows = group_rows_by_study_and_project
+    grouped_rows.each_value { |rows| process_rows(rows) }
+  end
+
+  private
+
+  def process_rows(rows)
+    barcodes = rows.pluck(headers.index(HEADER_BARCODE))
+    well_locations = rows.pluck(headers.index(HEADER_PLATE_WELLS))
+
+    if plate?(barcodes, well_locations)
```

Comment:
if it's not a plate or tube? e.g. we have tube racks now.
raise exception maybe

Reply:
plus add test for tube